Agenda for 0612 CG meeting (#665)
solid/specification: meetings/2024-06-12.md

ACTION: PAC to create a doodle.
  → https://evento.renater.fr/survey/solid-cg-meeting-at-tpac-2025-oat4znsh

* PAC: do we put all slots, or filter first?
* eP: prefer to add all and filter later
* PAC: maybe give a deadline.
* VB: (following minutes but cannot join): Our options are limited to what's available. Event will be added automatically to our calendar. Our first option of Monday was not available. Tuesday was not good for some people. Feel free to add all slots and we can ask what's possible but deadline was June 10th to submit, so our choices might no longer be available at this point. Let me know and I'll check. 
* eP: sharing with the group what VB is writing.
* PAC: we're past the deadline anyway
* eP: 48 hour should suffice. we cand send list mail and @room mention on matrix (any chair can do it)
* HZ: agreed 
* VB: Perhaps focus on Thu and Fri, since we already know Mon and Tue don't work. 

### initial WebID Document considerations

URL: https://github.com/solid/security-considerations/pull/9

* HZ: Jacopo is no longer a chair of WebID CG, not sure how it impacts Solid

> * eP: STM a week ago, JZ, VB, MDJ present. What's the criteria of accepting a proposal? For me, is to not conflict with the Solid protocol. Nobody showed me any problem with conformance. Discussing counter measures for attacks. My impression is that sometimes we put too much in a PR. If someone things something more needs to be added, they can create a new PR. IMHO this PR is ready to merge. If later we decide we can have more normative requirements, it can be changed.

Reviewer: ```suggestion
* eP: Solid Team Meeting a week ago; JZ, VB, MDJ present. What are the criteria for accepting a proposal? For me, it should not conflict with the Solid Protocol. Nobody showed me any problem with conformance. Discussing countermeasures for attacks. My impression is that sometimes we put too much in one PR. If someone thinks something more needs to be added, they can create a new PR. IMHO, this PR is ready to merge. If later we decide we want/need more normative requirements, it can be changed.
```

> * JZ: I objec to this PR. We have nothing in place to describe what a webid document does. I agree that we need to take into account security and keep webid docs outside solid servers is a valid option. Recommending that without anything else in place is only going to create a mess. I think it's premature to now start talking about where it leaves without first describing what a webid doc is. There is no consistency.
> * eP: I am aware of webid profile draft. Not ignoring that this work exist. Webid profile spec is not completely useless if webid is stored outside Solid. I have this diagram (screen shared). Solid protocol has a normative reference to the webid spec. I think we agreed last week that we need to agree about discovery, make parts public, some protected, but we're not there yet. I am having my WebID ouside the storage. It's a well established practice.
> * JZ: I am not trying to prevent people from having the document outside storage. I tried for 6 months to add a predicate for allowing a webid doc to be stored outside Solid. I think the content of webid should be established first. Discovery is important. I don't want to discuss it later. That extends the period of ambiguity for developers.

Reviewer: ```suggestion
* JZ: I object to this PR. We have nothing in place to describe what a WebID document does. I agree that we need to take security into account, and keeping WebID docs outside Solid servers is a valid option. Recommending that without anything else in place is only going to create a mess. I think it's premature to start talking about where it leaves without first describing what a WebID doc is. There is no consistency.
* eP: I am aware of WebID Profile draft. Not ignoring that this work exists. WebID Profile spec is not completely useless if WebID is stored outside Solid. I have this diagram (screen shared). Solid Protocol has a normative reference to the WebID spec. I think we agreed last week that we need to agree about discovery, make parts public, some protected, but we're not there yet. I have my WebID Document outside the Solid storage. It's a well established practice.
* JZ: I am not trying to prevent people from having their WebID Document outside Solid storage. I tried for 6 months to add a predicate for allowing a WebID Document to be stored outside Solid. I think the content of WebID Document should be established first. Discovery is important. I don't want to discuss it later. That extends the period of ambiguity for developers.
```

* eP: there are already at least 2 methods SAI and type indexes. This PR proposes a valid method. I think we have to have a method to separate concerns. I agree it took to long. I recognize the issue, but I don't think it's fair to block this PR. If you have a different option, please make a PR, you can make it option, WIP, and I am happy to add another option. I don't think we can wait adding a counter measure to a known security issue.
* JZ: Sarven and VB have similar objections to mine. If you can convince all of us, ok, but you didn't do that yet.

> * eP: maybe vote next week, try to reach consensus. SC and VB do not host webid doc in the solid server.

Reviewer 2: I do.

> * eP: there are already at least 2 methods SAI and type indexes. This PR proposes a valid method. I think we have to have a method to separate concerns. I agree it took to long. I recognize the issue, but I don't think it's fair to block this PR. If you have a different option, please make a PR, you can make it option, WIP, and I am happy to add another option. I don't think we can wait adding a counter measure to a known security issue.
> * JZ: Sarven and VB have similar objections to mine. If you can convince all of us, ok, but you didn't do that yet.
> * eP: maybe vote next week, try to reach consensus. SC and VB do not host webid doc in the solid server.

Reviewer: ```suggestion
* eP: there are already at least 2 methods: SAI, and Type Indexes (TIs). This PR proposes a valid method. I think we have to have a method to separate concerns. I agree it took to long. I recognize the issue, but I don't think it's fair to block this PR. If you have a different option, please make a PR; you can make it an option, WIP, and I am happy to add another option. I don't think we can wait to add a countermeasure to a known security issue.
* JZ: Sarven and VB have similar objections to mine. If you can convince all of us, OK, but you didn't do that yet.
* eP: Maybe vote next week, try to reach consensus. <strike>SC and</strike> VB does not host WebID Document in a Solid server.
```

> * TT: I think it makes some sense do divorce the webid doc from solid storage. Nothing says, that you MUST use webid as the identification methods. Focusing on webid puts roadblocks in our way. It's the only one listed, but it doesn't say it MUST be used.
> * eP: there was some conversation about using DIDs. I think it's opening a big can of worms. All implementations currently use WebID. If want to use something else too, we should provide such examples.
> * TT: DIDs certainly are a possibility. Security concerns should go againt Webid, not against Solid.
> * eP: It's not explicit to WebID, it's related to Solid's reliance on WebID. An attacker could inject something in the webid profile document and then get access to your data.

Reviewer: ```suggestion
* TT: I think it makes some sense to divorce the WebID doc from storage in a Solid server. Nothing says that you MUST use WebID as the identification method. Focusing on WebID puts roadblocks in our way. It's the only one listed, but it doesn't say it MUST be used.
* eP: There was some conversation about using DIDs. I think it's opening a big can of worms. All implementations currently use WebID. If we want to use something else too, we should provide such examples.
* TT: DIDs certainly are a possibility. Security concerns should go againt WebID, not against Solid.
* eP: It's not explicit to WebID; it's related to Solid's reliance on WebID. An attacker could inject something into the WebID Document and thereby get access to your data.
```

* TT: bad connection.
* TT: there is a circular reference. 

> * HZ: I agree and it should be priority, but we also are deviating from the topic. I also understand Jeff and it is also a priority. I also agree with Pavlik that what Jeff requests shouldn't block that PR. How do we move forward?

Reviewer: ```suggestion
* HZ: I agree that it should be a priority, but we are also deviating from the topic. I also understand Jeff and it is also a priority. I also agree with Pavlik that what Jeff requests shouldn't block that PR. How do we move forward?
```

* HZ: Jeff, Is it a strong objection or preference?
* JZ: I think Sarven and Virginia should be present. Some concerns need to be addressed.
* HZ: Do you agree that it is orthogonal to your issue?
* JZ: No I don't agree

> * HZ: From architecture point of view I agree with Pavlik. If you store WebID document on the Solid Pod, if pod is not available the webid document is also not available. But I'm not going to die on that hill.
> * JZ: There are 50k pods on solidcommunity.net those people and those pods I'm concern about. It is not orthogonal. There is an existing situation of knowledgable people having WebIDs out of the Solid server. At minimum Sarven and Virginia need to be present.

Reviewer: ```suggestion
* HZ: From an architecture point of view, I agree with Pavlik. If you store WebID Document on the Solid Pod, then when the pod is not available, the WebID Document is also not available. But I'm not going to die on that hill.
* JZ: There are 50k pods on solidcommunity.net. I'm concerned about those people and those pods. It is not orthogonal. There is an existing situation of knowledgeable people having WebIDs outside of the Solid server. At minimum, Sarven and Virginia need to be present.
```

* HZ: I agree with all of you, let's take it offline.

> * Ted: One of the challenges I had in PR9, the structure of the document needs to be more consumable and actionable. There is a blur between weaknesses and how to address them. The countermeasures are at the same level as the weaknesses. One is a theft of the token, there is another with theft of the credentials. There should be weakenss, attack and countermeasure that prevents it. And considerationsa about that weakenss. 
> * eP: It will be easier to structure based on more than one weakness.
> * HZ: I see more thought about infrastructure and not only focus on p2p

Reviewer: ```suggestion
* Ted: One of the challenges I had with PR9 is that the structure of the document needs to be more consumable and actionable. There is a blur between weaknesses and how to address them. The countermeasures section is at the same level as the individual weakness sections. One is a theft of the token; there is another with theft of credentials. There should be weakness, attack, and countermeasure that prevents it. And considerations about that weakness. 
* eP: It will be easier to structure based on more than one weakness.
* HZ: I see more thought about infrastructure and not only focus on p2p.
```

---

> * VB (following minutes but cannot join): Voting does not necessarily help with consensus. The issues with the PR are not objecting to the security consideration per se, but more seeking to add considerations with the approach. I am not opposed to hosting webid outside solid storage (obviously), but this PR seems to advocate for *not* hosting it on Solid storage, which is a strong message. I think we should look into how to securely allow to host. Happy to discuss next week / in another meeting. 

Reviewer: ```suggestion
* VB (following minutes but cannot join): Voting does not necessarily help with consensus. The issues with the PR are not objections to the security consideration, per se, but more seeking to add considerations to the approach. I am not opposed to hosting WebID Documents outside Solid servers (obviously), but this PR seems to advocate for *not* hosting it on Solid servers, which is a strong message. I think we should look into how to host securely. Happy to discuss next week / in another meeting. 
```

* VB: See: https://github.com/solid/security-considerations/pull/9#issuecomment-2155333752 on how to move forward.

Meeting ends

### N3 eligibility for being a normative reference in W3C WG deliverable

URL: https://solidproject.org/TR/protocol#normative-references
URL: https://w3c.github.io/N3/spec/

> https://github.com/solid/solid-wg-charter/issues/37#issuecomment-1580600098
> PAC: I think I remember @timbl talking about this not so long (during a Solid CG meeting??): as I remember, he was explaining that W3C RECs can depend on things that are one step behind in terms of maturity, precisely to make the co-evolution of standards more agile. I could not find anything about this in the current W3C process, though... But @csarven's example above shows there are precedents.

> * eP: We should clarify ASAP if N3 can be used as normative reference in any of WG deliverables. If it has to be one step behind in maturity we need to confirm if it is planned to be picked up by a WG in a near future and plan accordingly.

Reviewer: ```suggestion
* eP: We should clarify ASAP if N3 can be used as normative reference in Solid WG deliverables. If it can be no more than one step behind in maturity, we need to confirm whether it is planned to be picked up by a WG in the near future and plan accordingly.
```

> ### Proposed Work Item - Organizations
> 
> JZ: My prior work on organizations in Solid includes a dozen or so demos, blog posts, and github discussions going back to 2021.  These efforts lead to my role as lead of the Solid Team task force on converting solidproject.org to being generated from RDF. Denis Shirshov and Daniel Bakas have joined me in that task force.  At the suggestion of timbl, I created the github repo at https://github.com/solid/organizations with the eventual goal of developing a shape for organizational WebID profiles similar to the one he is creating for personal WebID profiles.   I will be proposing a new solid vocabulary specific to organizations and their resources, then developing shapes that use that and other vocabularies and eventually a possible client-to-client spec based on those.  I would like the assistance of the CG in involving more people in this process and in deciding where on github this should live.  My thinking is that we use the solid/organizations for discussion and development of the vocabuary and shapes and as the home of the client-to-client spec.  The vocabulary, when ready to be proposed would live in solid/vocab.  But I am open to suggestions.

Reviewer: ```suggestion
### Proposed Work Item — Organizations

JZ: My prior work on organizations in Solid includes a dozen or so demos, blog posts, and GitHub discussions going back to 2021.  These efforts led to my role as lead of the Solid Team task force on converting solidproject.org to being generated from RDF. Denis Shirshov and Daniel Bakas have joined me in that task force.  At the suggestion of TimBL, I created the [`solid/organizations` GitHub repo](https://github.com/solid/organizations) with the eventual goal of developing a shape for organizational WebID profiles, similar to the one he is creating for personal WebID profiles. I will be proposing a new solid vocabulary specific to organizations and their resources, then developing shapes that use that and other vocabularies and eventually a possible client-to-client spec based on those. I would like the assistance of the CG in involving more people in this process and in deciding where on GitHub this should live.  My thinking is that we use [`solid/organizations`](https://github.com/solid/organizations) for discussion and development of the vocabulary and shapes, and as the home of the client-to-client spec.  The vocabulary, when ready to be proposed, would live in [`solid/vocab`](https://github.com/solid/vocab).  But I am open to suggestions.
```

### Demo of solid efforts directory

URL: https://github.com/solid/specification/discussions/632
URL: https://elf-pavlik.github.io/solid-efforts | https://github.com/elf-pavlik/solid-efforts
URL: https://github.com/solid/specification/discussions/557#discussioncomment-6802365

> * eP: over the weekend I hand crafted a small dataset with initial information about people, drafts and implementations. Based on that data I'm making a very simple web app that attempts to show how things related by allowing to navigate it presented as linked tables. The dataset will be incrementally more inclusive and open to PRs.

Reviewer: ```suggestion
* eP: Over the weekend, I handcrafted a small dataset with initial information about people, drafts, and implementations. Based on that data, I'm making a very simple web app that attempts to show how things are related by presenting them as linked tables, and allowing navigation through them. The dataset will be incrementally more inclusive and open to PRs.
```

![image](https://hackmd.io/_uploads/BJRJkFEH0.png)
![image](https://hackmd.io/_uploads/HJeNOmDSA.png)
![localhost_3000_ (1)](https://hackmd.io/_uploads/r1_yqpHHA.png)

* SC: While I appreciate the enthusiasm behind "Solid Efforts", I find it disjoint from existing efforts in the Solid community where the underlying data is actually authoritative (source) and structured (and queryable), e.g.:
  * Solid Technical Reports: https://solidproject.org/TR/
  * Solid Project's listings of servers and implementations, e.g., https://solidproject.org/apps
  * Solid QA: https://solidproject.org/ED/qa
* SC: Whereas manually created (opinionated) data duplicates data/efforts and will definitely have errors and cause unnecessary further confusion.

> * SC: So, I'd recommend putting energy on improving data quality (and structure/semantics) at *source* and building applications using that data.

Reviewer: ```suggestion
* SC: So, I'd recommend putting energy into improving data quality (and structure/semantics) at *source*, and building applications using that data.
```

> * SC: Some of the technical reports can already be serialized in RDF, and so they can be consumed/reused in variety ways (that's partly what Solid QA is doing.)

Reviewer: ```suggestion
* SC: Some of the technical reports can already be serialized in RDF, so they can be consumed/reused in a variety of ways. (That's partly what Solid QA is doing.)
```
